static/js/components/PostDisplay.js

```diff
@@ -23,30 +27,41 @@ const postTitle = (post: Post) =>
       {post.title}
     </a>
 
-export default class PostDisplay extends React.Component {
+class PostDisplay extends React.Component {
   props: {
     post: Post,
     showChannelLink?: boolean,
-    expanded?: boolean
+    expanded?: boolean,
+    dispatch: Dispatch
   }
 
   showChannelLink = () => {
     const { post, showChannelLink } = this.props
 
-    return showChannelLink && post.channel_name
+    return (showChannelLink && post.channel_name
       ? <Link to={`/channel/${post.channel_name}`}>
           on {post.channel_name}
       </Link>
-      : null
+      : null)
+  }
+
+  toggleUpvote = () => {
+    const { dispatch, post } = this.props
+    dispatch(actions.postUpvotes.post(post.id, !post.upvoted)).then(post => {
+      dispatch(setPostData(post))
+    })
```

Comment:
@gsidebo we need `toggleUpvote` in `Postpage`, `ChannelPage`, and `Homepage` so that's why we probably want to pull the impl out into a util.